utils/src/main/kotlin/Utils.kt

```diff
@@ -175,11 +175,27 @@ fun getPathFromEnvironment(executable: String): File? {
     val paths = Os.env["PATH"]?.splitToSequence(File.pathSeparatorChar).orEmpty()
 
     return if (Os.isWindows) {
+        val nestedVariableRegex = Regex("%(?<reference>\\w+)%")
+
         paths.mapNotNull { path ->
-            resolveWindowsExecutable(File(path, executable))
+            val expandedPath = path.replace(nestedVariableRegex) {
+                val variableReference = it.groups["reference"]!!.value
+                Os.env[variableReference] ?: variableReference
+            }
+
+            resolveWindowsExecutable(File(expandedPath, executable))
         }.firstOrNull()
     } else {
-        paths.map { path -> File(path, executable) }.find { it.isFile }
+        val nestedVariableRegex = Regex("\\$\\{?(?<reference>\\w+)}?")
+
+        paths.map { path ->
+            val expandedPath = path.replace(nestedVariableRegex) {
```

Comment:
There is some redundancy here. The replacement of variables in the path could be done in a helper function that takes the regular expression as a parameter.